lib/core/src/Cardano/Wallet/Types/Read/Tx.hs

```diff
@@ -74,11 +74,29 @@ but the type family `Api.ShelleyLedgerEra` performs the conversion.
 -- This transaction can come from any era, old (Byron) and new (Babbage).
 data Tx where
     Tx
-        :: forall era. Api.IsCardanoEra era
+        :: forall era
+        . Api.IsCardanoEra era
         => Api.CardanoEra era
         -> TxEra era
         -> Tx
 
+instance Show Tx where
+    show (Tx ByronEra e) = show e
+    show (Tx ShelleyEra e) = show e
+    show (Tx AllegraEra e) = show e
+    show (Tx MaryEra e) = show e
+    show (Tx AlonzoEra e) = show e
+    show (Tx BabbageEra e) = show e
+
+instance Eq Tx where
+    Tx ByronEra x == Tx ByronEra x' = x == x'
+    Tx ShelleyEra x == Tx ShelleyEra x' = x == x'
+    Tx AllegraEra x == Tx AllegraEra x' = x == x'
+    Tx MaryEra x == Tx MaryEra x' = x == x'
+    Tx AlonzoEra x == Tx AlonzoEra x' = x == x'
+    Tx BabbageEra x == Tx BabbageEra x' = x == x'
+    _ == _ = False
```

Comment:
Note that due to the usage of a wildcard, the compiler will not warn us to update this function if another era is added. See [Coding-Standards:avoid-wildcards-when-pattern-matching-on-sum-types](https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md#avoid-wildcards-when-pattern-matching-on-sum-types).

I don't know of a nice way to auto-derive `Eq` instances for existential types like this one.

However, I can think of two ways to write an `Eq` instance that survives the addition of further eras:

1. Use the fact that we already have a `Show` instance:
    ```hs
      instance Eq Tx where
          (==) = (==) `on` show
    ```
    Assuming the laziness of `show`, this should short-circuit as soon as there's a character mismatch. On the other hand, if both sides are equal, then we'll have to fully evaluate both sides, which might give poor performance. 👎🏻 

2. Add `Typeable era` and `Typeable (TxEra era)` constraints to the `era` parameter of the `Tx` constructor. Then use [`cast`](https://hackage.haskell.org/package/base-4.15.0.0/docs/Data-Typeable.html#v:cast):
    ```hs
    instance Eq Tx where
        Tx e1 t1 == Tx e2 t2 = (&&)
            (Just e1 == cast e2)
            (Just t1 == cast t2)
    ```

Reply:
If neither of the above options seem palatable, then you could still avoid the wildcard with:

```hs
instance Eq Tx where
    (==) (Tx ByronEra   x) = \case {Tx ByronEra   y -> x == y; _ -> False}
    (==) (Tx ShelleyEra x) = \case {Tx ShelleyEra y -> x == y; _ -> False}
    (==) (Tx AllegraEra x) = \case {Tx AllegraEra y -> x == y; _ -> False}
    (==) (Tx MaryEra    x) = \case {Tx MaryEra    y -> x == y; _ -> False}
    (==) (Tx AlonzoEra  x) = \case {Tx AlonzoEra  y -> x == y; _ -> False}
    (==) (Tx BabbageEra x) = \case {Tx BabbageEra y -> x == y; _ -> False}
```

Reply:
thanks